tee-worker/litentry/pallets/identity-management/src/identity_context.rs

```diff
@@ -101,10 +103,26 @@ pub fn get_eligible_identities<T: Config>(
 				//   care about web2 identities, this step will empty `IdentityContext.web3networks`
 				// - it helps to reduce the request size a bit
 				networks.retain(|n| desired_web3networks.contains(n));
-				if networks.is_empty() && item.0.is_web3() {
-					None
+
+				if force_retain_web2_identity && item.0.is_web2() {
+					return Some((item.0.clone(), Vec::new()))
+				}
+
+				// differentiate between web2 and web3 assertions:
+				// desired_web3networks.is_empty() means it's a web2 assertion,
+				// otherwise web2 identities might survive to be unexpectedly "eligible" for web3 assertions.
+				if desired_web3networks.is_empty() {
+					if item.0.is_web2() {
+						Some((item.0.clone(), networks))
+					} else {
+						None
+					}
 				} else {
-					Some((item.0.clone(), networks))
+					if item.0.is_web3() && !networks.is_empty() {
+						Some((item.0.clone(), networks))
+					} else {
+						None
+					}
```

Comment:
Maybe use below code to simplify the nesting levels of if-else:
```
if (force_retain_web2_identity || desired_web3networks.is_empty()) && item.0.is_web2() {
	Some((item.0.clone(), networks))
} else if item.0.is_web3() && !networks.is_empty() {
	Some((item.0.clone(), networks))
} else {
	None
}
```

Reply:
Thanks!

I intentionally keep the nested structure to (hopefully) retain the readability. While it can be logiacally further simplified, down goes the readability too (IMO).

So I'll keep it, it's probably a style problem.